app/src/main/kotlin/compiler/ast/ExpressionAst.kt

```diff
@@ -0,0 +1,63 @@
+package compiler.ast
+
+sealed class ExpressionAst {
+    object UnitLiteral : ExpressionAst()
+    data class BooleanLiteral(val value: Boolean) : ExpressionAst()
+    data class NumberLiteral(val value: Int) : ExpressionAst()
+
+    data class Variable(val name: String) : ExpressionAst()
+
+    data class FunctionCall(
+        val name: String,
+        val arguments: List<Argument>
+    ) : ExpressionAst() {
+        data class Argument(
+            val name: String?,
+            val value: ExpressionAst
+        )
+    }
+
+    data class UnaryOperation(
+        val kind: Kind,
+        val operand: ExpressionAst
+    ) : ExpressionAst() {
+        enum class Kind {
+            NOT,
+            NEG,
+            BIT_NOT,
+        }
```

Comment:
Just to confirm: we don't want to have an unary `plus`, right?

Reply:
We can add it, but I suppose the language won't become unusable without it.

Reply:
Added, and made the operation names more readable.